### PR TITLE
Add docs for Azure and Database Shells Configuration

### PR DESCRIPTION
**Pull request by deanmarcussen**

### CODE REVIEW

File: src/docs/reference/core/Shells/README.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+The primary pupose of the Shell Configuration providers is to provide a shared external environment for multi-tenancy
+where tenants need to be created, and their settings mutated, during live operation of the stateless hosts.
+
+It is not intended to support shared configuration between local development and production environments.
+
+## Azure Shells Configuration Provider
+
+The Azure Shells Configuration provider uses an Azure Blob Storage Container to store the `tenants.json` and related tenant `appsettings.json` 
+files in a similar hierachy to that of the default `App_Data` configuration.
```
Comment (agriffard) [inline src/docs/reference/core/Shells/README.md:21]: hierachy => hierarchy

File: src/docs/reference/core/Shells/README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+The root of the Azure Blob Container includes a `tenants.json` file, and optionally can include a `appsettings.json` file.
+
+Each shell, or tenant has a directory under the `Sites` folder, named for the tenant, with an individual `appsettings.json file.
```
Comment (agriffard) [inline src/docs/reference/core/Shells/README.md:25]: missing closing `

File: src/docs/reference/core/Shells/README.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+The hierachy is seperated into single files, and is useful if you need to manage the tenants `appsettings.json` independently from Orchard Core
```
Comment (agriffard) [inline src/docs/reference/core/Shells/README.md:27]: hierachy => hierarchy

File: src/docs/reference/core/Shells/README.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+}
+```
+
+!!! note
+    The database must be created before using the Database Configuration provider.
+    Make sure this database is secure and access to it is limited.
+
+## Migrate From Files
+
+The `MigratFromFiles` option is available for both the Azure Shells and Database Shells Configuration providers
```
Comment (agriffard) [inline src/docs/reference/core/Shells/README.md:120]: MigratFromFiles => MigrateFromFiles

File: src/docs/reference/core/Shells/README.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+When enabled the Shell Configuration provider will first check to see if a configuration exists for a given tenant
+for the chosen storage platform, Database, or Azure Blob Storage.
+
+If the configuration does not exist, the provider will try to load it from the `App_Data` folder, 
+and migrate it to the storage platform of choice.
+
+## Enviroment Options
+
+To disabled a provider in Development, or different environments inject the `IHostEnvironment`
```
Comment (agriffard) [inline src/docs/reference/core/Shells/README.md:131]: disabled => disable

Comment (agriffard) [inline src/docs/reference/core/Shells/README.md:131]: different environments inject => different environments, inject

Comment (deanmarcussen) [reply src/docs/reference/core/Shells/README.md]: Thanks @agriffard spelling fixed

**Comment by deanmarcussen** [comment]
Thanks @agriffard fixed spelling

**Pull request merged**